### ISSUE TITLE
v6 alpha 14: Cannot read property 'map' of undefined

### ISSUE THREAD

**Issue by aight8** [erikras/redux-form#1040]
Got following error on submit since version v16 alpha 14:

<img width="614" alt="bildschirmfoto 2016-05-25 um 03 50 35" src="https://cloud.githubusercontent.com/assets/1138623/15525622/f172415c-222b-11e6-9ff5-b8dd28253a91.png">
<img width="731" alt="bildschirmfoto 2016-05-25 um 03 53 02" src="https://cloud.githubusercontent.com/assets/1138623/15525653/37a05c9a-222c-11e6-9321-e472c8b37047.png">

**Comment by erikras** [comment]
Are you seeing the `REGISTER_FIELD` actions dispatched in Redux?

**Comment by aight8** [comment]
Yep It appears for every field which is rendered. Even when I modify the form it unregister or register some fields. But when I submit this error appears.

**Comment by ooflorent** [comment]
Could you create a gist or something else so we could easily reproduce the issue?

**Comment by pr0digy** [comment]
This seems to happen when initalValue is first set to undefined, before async action loads data, and then after the data is loaded and available the above error occurs on submit.

**Comment by Iuriy-Budnikov** [comment]
The same issue +1

**Comment by Iuriy-Budnikov** [comment]
@ooflorent i got this issue when trying submit -> reset -> submit

**Comment by clayne11** [comment]
Please create a repro and I will this issue.

**Comment by omgaz** [comment]
I'm noticing the exact same issue with `wrapped of undefined`. 

Main.js
```es6
import { reducer as reduxFormReducer } from 'redux-form';
import { ExampleForm } from 'ExampleFormComponent';
const reducer = combineReducers({
  visibility: visibilityReducer,
  form: reduxFormReducer,
});
```

ExampleFormComponent.js
```es6
import React, { PropTypes } from 'react';
import { Field, reduxForm } from 'redux-form';

const ExampleForm = (props) => {
  const { handleSubmit, reset, submitting } = props;
  return (
    <form className="form-horizontal" onSubmit={handleSubmit}>
        <div>
          <fieldset>
            <Field name="firstName" component="input" type="text" placeholder="First Name"/>
          </fieldset>
        </div>
    </form>
  );
};

ExampleForm.propTypes = {
  reset: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  submitting: PropTypes.bool.isRequired,
};

export default reduxForm({
  form: 'example',
})(ExampleForm);
```

**Comment by rafsoaken** [comment]
I downgraded to 13 because of this issue - happened whenever i tried to dispatch the "initialize" action creator after receiving async data.

**Comment by clayne11** [comment]
Please post a reproduction repo that I can clone and I'll happily fix it.

**Comment by babsonmatt** [comment]
@clayne http://redux-form.com/6.0.0-alpha.14/examples/submitValidation/

**Comment by clayne11** [comment]
Thanks - I'll have a look at it this weekend.

**Comment by erikras** [comment]
No need, @clayne11. I got it.

Will leave another comment here when it's released in the next version.

**Comment by JakeLin1024** [comment]
I got this issue too when trying submit -> initialize -> submit

**Comment by erikras** [comment]
Good one, @JakeLin1024. I just fixed that case, too.

**Comment by Iuriy-Budnikov** [comment]
@erikras , Could you please release this version?

**Comment by erikras** [comment]
Published as [`6.0.0-alpha-15`](https://github.com/erikras/redux-form/releases/tag/v6.0.0-alpha.15).
